contrib/quill-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,106 @@
+set (LIBRARY_DIR "${ClickHouse_SOURCE_DIR}/contrib/quill")
+
+add_library(_quill INTERFACE)
+
+target_include_directories(_quill SYSTEM INTERFACE "${LIBRARY_DIR}/include")
+
+add_library(ch_contrib::quill ALIAS _quill)
+
+# header files
+set(HEADER_FILES
+        ${LIBRARY_DIR}/include/quill/backend/BackendManager.h
+        ${LIBRARY_DIR}/include/quill/backend/BackendOptions.h
+        ${LIBRARY_DIR}/include/quill/backend/BackendWorker.h
+        ${LIBRARY_DIR}/include/quill/backend/BacktraceStorage.h
+        ${LIBRARY_DIR}/include/quill/backend/ManualBackendWorker.h
+        ${LIBRARY_DIR}/include/quill/backend/PatternFormatter.h
+        ${LIBRARY_DIR}/include/quill/backend/RdtscClock.h
+        ${LIBRARY_DIR}/include/quill/backend/SignalHandler.h
+        ${LIBRARY_DIR}/include/quill/backend/StringFromTime.h
+        ${LIBRARY_DIR}/include/quill/backend/ThreadUtilities.h
+        ${LIBRARY_DIR}/include/quill/backend/TimestampFormatter.h
+        ${LIBRARY_DIR}/include/quill/backend/TransitEvent.h
+        ${LIBRARY_DIR}/include/quill/backend/TransitEventBuffer.h
+        ${LIBRARY_DIR}/include/quill/backend/BackendUtilities.h
+
+        ${LIBRARY_DIR}/include/quill/bundled/fmt/args.h
```

Comment:
I wonder if this won't cause issues with our own bundled fmt if/when the versions don't match exactly

Reply:
We use custom formatters that AFAIK we format everything before passing it to Quill (at least we do for other loggers). But maybe @antonio2368 can confirm this is still the case.

Reply:
fmt is still used in background thread regardless where we do the main formatting.